api/disabled_parameters.go

```diff
@@ -39,18 +42,171 @@ func NewDisabledMap() *DisabledMap {
 	}
 }
 
+// DisabledMapConfig is a type that holds the configuration for setting up
+// a DisabledMap
+type DisabledMapConfig struct {
+	// Key -> Path of REST endpoint (i.e. /v2/accounts/{account-id}/transactions)
+	// Value -> Operation "get, post, etc" -> Sub-value: List of parameters disabled for that endpoint
+	Data map[string]map[string][]string
+}
+
+// IsDisabled Returns true if the parameter is disabled for the given path
+func (dmc *DisabledMapConfig) IsDisabled(restPath string, operationName string, parameterName string) bool {
+	parameterList, exists := dmc.Data[restPath][operationName]
+	if !exists {
+		return false
+	}
+
+	for _, parameter := range parameterList {
+		if parameterName == parameter {
+			return true
+		}
+	}
+	return false
+}
+
+func (dmc *DisabledMapConfig) addEntry(restPath string, operationName string, parameterNames []string) {
+
+	if dmc.Data == nil {
+		dmc.Data = make(map[string]map[string][]string)
+	}
+
+	if dmc.Data[restPath] == nil {
+		dmc.Data[restPath] = make(map[string][]string)
+	}
+
+	dmc.Data[restPath][operationName] = parameterNames
+}
+
+// NewDisabledMapConfig creates a new disabled map configuration
+func NewDisabledMapConfig() *DisabledMapConfig {
+	return &DisabledMapConfig{
+		Data: make(map[string]map[string][]string),
+	}
+}
+
+// GetDefaultDisabledMapConfigForPostgres will generate a configuration that will block certain
+// parameters.  Should be used only for the postgres implementation
+func GetDefaultDisabledMapConfigForPostgres() *DisabledMapConfig {
+	rval := NewDisabledMapConfig()
+
+	// Some syntactic sugar
+	get := func(restPath string, parameterNames []string) {
+		rval.addEntry(restPath, http.MethodGet, parameterNames)
+	}
+
+	get("/v2/accounts", []string{"currency-greater-than", "currency-less-than"})
+	get("/v2/accounts/{account-id}/transactions", []string{"note-prefix", "tx-type", "sig-type", "asset-id", "before-time", "after-time", "rekey-to"})
+	get("/v2/assets", []string{"name", "unit"})
+	get("/v2/assets/{asset-id}/balances", []string{"round", "currency-greater-than", "currency-less-than"})
+	get("/v2/transactions", []string{"note-prefix", "tx-type", "sig-type", "asset-id", "before-time", "after-time", "currency-greater-than", "currency-less-than", "address-role", "exclude-close-to", "rekey-to", "application-id"})
+	get("/v2/assets/{asset-id}/transactions", []string{"note-prefix", "tx-type", "sig-type", "asset-id", "before-time", "after-time", "currency-greater-than", "currency-less-than", "address-role", "exclude-close-to", "rekey-to"})
+
+	return rval
+}
+
+// ErrDisabledMapConfig contains any mis-spellings that could be present in a configuration
+type ErrDisabledMapConfig struct {
```

Comment:
I think a lot of these can be private, the errors interface should be sufficient for public usage.